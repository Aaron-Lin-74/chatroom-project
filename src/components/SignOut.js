import { isUserSignedIn, signOutUser } from '../firebase'

function SignOut() {
  return (
    isUserSignedIn() && (
      <button className='sign-out' onClick={signOutUser}>
        SIGN-OUT
      </button>
    )
  )
}

export default SignOut
