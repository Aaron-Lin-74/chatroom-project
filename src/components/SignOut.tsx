function SignOut({ signOutUser }: { signOutUser: () => void }) {
  return (
    <button type='button' className='sign-out chat-btns' onClick={signOutUser}>
      SIGN-OUT
    </button>
  );
}

export default SignOut;
