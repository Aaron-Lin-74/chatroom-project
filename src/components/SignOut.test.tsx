import SignOut from './SignOut';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Test suites for SignOut component', () => {
  test('should match the snapshot', () => {
    const { asFragment } = render(<SignOut signOutUser={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render the component', () => {
    render(<SignOut signOutUser={() => {}} />);
    // screen.debug();
    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent('SIGN-OUT');
  });

  test('should handle onClick', () => {
    const onClick = jest.fn();
    render(<SignOut signOutUser={onClick} />);
    const buttonElement = screen.getByRole('button');
    fireEvent.click(buttonElement);
    expect(onClick).toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
