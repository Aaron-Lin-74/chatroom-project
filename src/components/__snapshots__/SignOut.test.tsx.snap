// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suites for SignOut component should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sign-out chat-btns"
    type="button"
  >
    SIGN-OUT
  </button>
</DocumentFragment>
`;
